### ISSUE TITLE
Broken code samples in "Defending Pyramid's Design" document

### ISSUE THREAD

**Issue by roganov** [Pylons/pyramid#2287]
```python
app.route('/:action')
def action():
    if action == 'add':
       return '<html>add</html>'
    if action == 'delete':
       return '<html>delete</html>'
    return app.abort(404)
```
As you can see this function compares function `action` to strings "add" and "delete".

**Comment by stevepiercy** [comment]
This code should be replaced with a working example that demonstrates ordering of routes. This sample was added about 5 years ago and updated for Pyramid 1.1 at the time. See https://github.com/Pylons/groundhog

Please submit a working example, sign CONTRIBUTORS.txt, and submit a PR, so we can review and test it.

**Comment by mmerickel** [comment]
It just needs to be `def action(action):` to work.

**Comment by stevepiercy** [comment]
@mmerickel for context, the full code sample is here:
http://docs.pylonsproject.org/projects/pyramid/en/latest/designdefense.html#routes-need-relative-ordering

**Comment by stevepiercy** [comment]
I gave it my best shot. I cloned groundhog, created a venv, edited setup.py to pin versions back to that which existed 5 years ago:

```python
install_requires=[
    'pyramid==1.1',
    'pyramid_jinja2<=1.0',
    'Paste<=1.7.5.1',
    'WebError<=0.10.3',
    ]
```
Installed Pyramid, etc., and ran the sample app from the docs link above.
```bash
$ $VENV/bin/python setup.py develop
$ $VENV/bin/python hello.py
```

Visited http://127.0.0.1:8080/admin

> 404 Not Found
> The resource could not be found.
> /admin

In Pyramid 1.6, there's no globals for config, which is why I tried going back in time.

I'm tempted to steal from the chapter on URL Dispatch, [Route Declaration Ordering](http://docs.pylonsproject.org/projects/pyramid/en/latest/narr/urldispatch.html#route-declaration-ordering).
